### DISCUSSION TITLE
Downloading of Album takes long time without any indication, that stuff is happening?

### DISCUSSION THREAD

**Discussion by tillhanke** [photoprism/photoprism#4056] [Q&A]
Hi there, I just created a album of photos and small videos of the last year and wanted to download that. When clicking on the download button in album view a short pop up banner appears and tells me that the download is starting. Then nothing happens.

I thought something had gone wrong and looked into the logs. To find out that it in fact was collection photos for the download. But it takes LONG. about 2-5s per picture. And the worst is, I can not find any indication on the website, that actually something is happening in the background. Am I missing something? Or might this be something to be improved in the future?

**Comment by graciousgrey** [comment]
Many thanks for your feedback! 2-5s per image seems very long, on our private installations it is much shorter. However, the download speed may depend among other factors on the individual setup/file system or the size of the downloaded album. 

Currently you can find information about a running download in the logs (/library/logs) and also the browser should show that a download is running (at least Chrome and Firefox do). We could add more indicators, however, this is not a high priority compared to the other tasks on our to-do list.

**Reply by lastzero** [reply]
It might help the performance if we reduce the zip compression level as most image formats are already highly compressed. However, when the compression / CPU isn't the bottleneck then it could also be an I/O issue, for example slow network storage or very large files. A persistent or larger notification may help in this case, but still won't make things any faster than the network/hardware allows.

**Reply by tillhanke** [reply]
The problem is, that the preparation of the download takes long. Maybe I configured something wrong... 
It adds photos to the zip and then after the 100th or so the following error line appears:
```
{"log":"time=\"2024-02-14T20:47:17Z\" level=error msg=\"download: failed adding 2023/10/20231011_062746_F7CFA382.heic.jpg to album zip (write tcp 172.25.0.4:2342-\u003e172.25.0.2:47120: write: connection reset by peer)\"\n","stream":"stderr","time":"2024-02-14T20:47:17.19132882Z"}
```
The CPU doesn't show any major activity. 
The network shouldn't be a problem (even though it is pretty slow only a german internet connection...), as for my understanding it is slow preparing the zip file. Or am I wrong there?

**Reply by lastzero** [reply]
When you download complete albums with a recent release, the zip archives are sent to the browser without creating any temporary files. So there shouldn't be much to prepare unless you didn't select a single album for download but individual/multiple files? In that case, a temporary zip file is created. However, this should not cause errors to be logged either unless e.g. the disk is full or there are filesystem permission problems. You could reduce the log level for troubleshooting if you haven't done that yet.

**Reply by lastzero** [reply]
Also make sure your index is up to date in case any filenames have changed or files have been deleted.

**Reply by m00nwtchr** [reply]
Also getting this exact error, it might have something to do with running photoprism over a `cloudflared` tunnel? (If that is what OP is also using)

This only happens when downloading a large number of pictures though (each about 8-10 MB), like a whole album of 300, but not only some of the files from the same album.
```
[A number of successful "download: added XXX" lines, and then:]
photoprism-1   | time="2024-05-20T16:38:59Z" level=error msg="download: failed adding 2024/05/20240519_144056_97C5B856.jpg to album zip (write tcp 172.25.0.4:2342->172.25.0.2:37714: write: connection reset by peer)"
cloudflared-1  | 2024-05-20T16:38:59Z ERR  error="timeout: no recent network activity" connIndex=1 event=1 ingressRule=1 originService=http://photoprism:2342
cloudflared-1  | 2024-05-20T16:38:59Z INF Unregistered tunnel connection connIndex=1 event=0 ip=XXX.XXX.XXX.XXX
cloudflared-1  | 2024-05-20T16:38:59Z ERR Request failed error="timeout: no recent network activity" connIndex=1 dest=https://XXXXXXXXXXXX.XXX/api/v1/albums/XXXXXX/dl?t=XXXXXX event=0 ip=XXX.XXX.XXX.XXX type=http
cloudflared-1  | 2024-05-20T16:38:59Z WRN Failed to serve quic connection error="context canceled" connIndex=1 event=0 ip=198.41.200.23
cloudflared-1  | 2024-05-20T16:38:59Z INF Retrying connection in up to 1s connIndex=1 event=0 ip=XXX.XXX.XXX.XXX
cloudflared-1  | 2024-05-20T16:39:00Z WRN Connection terminated error="context canceled" connIndex=1
```
Seems to me like maybe cloudflare is reading the long reply as "idle" and shutting down the connection.
